In [1]:
from pathlib import Path
import pandas as pd

project_root = Path(r"C:\Users\HP-ZBOOK i7\graphrag_test")

pilot1_reports = pd.read_csv(
    project_root
    / "pilot_01"
    / "inspection"
    / "07_community_reports_all.csv"
)

pilot2_reports = pd.read_csv(
    project_root
    / "pilot_02"
    / "inspection"
    / "07_community_reports_all.csv"
)

print(
    "Pilot 1 community reports:",
    len(pilot1_reports),
)

print(
    "Pilot 2 community reports:",
    len(pilot2_reports),
)

absolute_change = (
    len(pilot2_reports)
    - len(pilot1_reports)
)

percentage_change = (
    absolute_change
    / len(pilot1_reports)
    * 100
)

print("Absolute change:", absolute_change)
print(
    "Percentage change:",
    round(percentage_change, 1),
    "%",
)

print("\nPilot 2 columns:")
print(pilot2_reports.columns.tolist())

Pilot 1 community reports: 62
Pilot 2 community reports: 110
Absolute change: 48
Percentage change: 77.4 %

Pilot 2 columns:
['id', 'human_readable_id', 'community', 'level', 'parent', 'children', 'title', 'summary', 'full_content', 'rank', 'rating_explanation', 'findings', 'full_content_json', 'period', 'size']


In [2]:
top_ranked_reports = pilot2_reports.nlargest(
    5,
    "rank",
)

largest_reports = pilot2_reports.nlargest(
    5,
    "size",
)

random_reports = pilot2_reports.sample(
    n=7,
    random_state=42,
)

community_report_sample = (
    pd.concat(
        [
            top_ranked_reports,
            largest_reports,
            random_reports,
        ]
    )
    .drop_duplicates(subset="id")
    .reset_index(drop=True)
)

print(
    "Community-report audit sample size:",
    len(community_report_sample),
)

community_report_sample[
    [
        "community",
        "level",
        "title",
        "summary",
        "full_content",
        "rank",
        "size",
    ]
]

Community-report audit sample size: 13


,community,level,title,summary,full_content,rank,size
0,105,2,Austrian Renewable Expansion Act (EAG) – PV Ex...,This community centers on the Erneuerbaren-Aus...,# Austrian Renewable Expansion Act (EAG) – PV ...,9.0,6
1,76,1,Photovoltaics and Systemic Expansion in Austri...,This community covers the extensive role and s...,# Photovoltaics and Systemic Expansion in Aust...,9.0,38
2,12,0,"Support Schemes, Legal Targets, and Constraint...",The supplied graph records describe interconne...,"# Support Schemes, Legal Targets, and Constrai...",9.0,34
3,18,0,Photovoltaics and Sectoral Integration in the ...,This community centers on photovoltaics (PV) a...,# Photovoltaics and Sectoral Integration in th...,9.0,62
4,19,0,Austrian Photovoltaic Strategy and Its Ecosystem,This community is centered on the Austrian Pho...,# Austrian Photovoltaic Strategy and Its Ecosy...,9.0,34
5,9,0,Austrian Photovoltaic Expansion Targets and Su...,This community centers on Austria's scenario-b...,# Austrian Photovoltaic Expansion Targets and ...,8.0,49
6,3,0,"PV Acceptance, Biodiversity, and Workforce Dev...",This graph community centers on interlinked th...,"# PV Acceptance, Biodiversity, and Workforce D...",8.5,38
7,91,1,Austrian Photovoltaics Research and Innovation...,This community describes the interconnected la...,# Austrian Photovoltaics Research and Innovati...,5.5,3
8,23,1,"Stakeholders, Action Fields, and Strategic Aim...",The supplied graph records describe the Austri...,"# Stakeholders, Action Fields, and Strategic A...",7.5,12
9,97,1,Regulatory Reform for PV Integration in Austri...,This community is centered on the ELEKTRIZITÄT...,# Regulatory Reform for PV Integration in Aust...,8.0,10


In [3]:
community_report_review = community_report_sample[
    [
        "id",
        "community",
        "level",
        "title",
        "summary",
        "full_content",
        "rank",
        "size",
    ]
].copy()

community_report_review["source_grounding"] = ""
community_report_review["primary_classification"] = ""
community_report_review["inherited_graph_error"] = ""
community_report_review["new_llm_claim"] = ""
community_report_review["numerical_claims_correct"] = ""
community_report_review["modality_preserved"] = ""
community_report_review["causal_language_problem"] = ""
community_report_review["unsupported_elaboration"] = ""
community_report_review["audit_notes"] = ""

community_report_review.head()

,id,community,level,title,summary,full_content,rank,size,source_grounding,primary_classification,inherited_graph_error,new_llm_claim,numerical_claims_correct,modality_preserved,causal_language_problem,unsupported_elaboration,audit_notes
0,71b22030b1d7b0366d0f5f42e7c99e6d8228cc6b220605...,105,2,Austrian Renewable Expansion Act (EAG) – PV Ex...,This community centers on the Erneuerbaren-Aus...,# Austrian Renewable Expansion Act (EAG) – PV ...,9.0,6,,,,,,,,,
1,5059b9ee7ff85ddc622abdc4053f2bdf53c8e1f94e35a8...,76,1,Photovoltaics and Systemic Expansion in Austri...,This community covers the extensive role and s...,# Photovoltaics and Systemic Expansion in Aust...,9.0,38,,,,,,,,,
2,8632dc2b59a63ffdb446755a0b6cc3e72f368578230f5e...,12,0,"Support Schemes, Legal Targets, and Constraint...",The supplied graph records describe interconne...,"# Support Schemes, Legal Targets, and Constrai...",9.0,34,,,,,,,,,
3,f9ecf79cae6f90e7dcb23022e51fb14d65d02a75c6a71a...,18,0,Photovoltaics and Sectoral Integration in the ...,This community centers on photovoltaics (PV) a...,# Photovoltaics and Sectoral Integration in th...,9.0,62,,,,,,,,,
4,67492b2814bbce24cad9e7a687c639724cc866bc2bf14d...,19,0,Austrian Photovoltaic Strategy and Its Ecosystem,This community is centered on the Austrian Pho...,# Austrian Photovoltaic Strategy and Its Ecosy...,9.0,34,,,,,,,,,


In [4]:
sample_path = (
    project_root
    / "pilot_02"
    / "inspection"
    / "13_community_report_review_sample.csv"
)

community_report_review.to_csv(
    sample_path,
    index=False,
    encoding="utf-8-sig",
)

print("Community-report review sample saved to:")
print(sample_path)

Community-report review sample saved to:
C:\Users\HP-ZBOOK i7\graphrag_test\pilot_02\inspection\13_community_report_review_sample.csv


In [5]:
community_audit = {
    105: {
        "source_grounding": "HIGH",
        "primary_classification": "ACCURATE_SYNTHESIS",
        "inherited_graph_error": "NO",
        "new_llm_claim": "NO",
        "numerical_claims_correct": "YES",
        "modality_preserved": "YES",
        "causal_language_problem": "NO",
        "unsupported_elaboration": "NO",
        "audit_notes": (
            "The EAG targets of 100% balance-sheet renewable electricity "
            "consumption and at least 11 TWh additional PV generation by "
            "2030 are represented correctly. The report also distinguishes "
            "targets and support mechanisms from observed outcomes."
        ),
    },

    76: {
        "source_grounding": "PARTIAL",
        "primary_classification": "ACCURATE_SYNTHESIS",
        "inherited_graph_error": "YES",
        "new_llm_claim": "NO",
        "numerical_claims_correct": "NOT_APPLICABLE",
        "modality_preserved": "YES",
        "causal_language_problem": "NO",
        "unsupported_elaboration": "NO",
        "audit_notes": (
            "The report accurately synthesizes many PV policies, targets, "
            "constraints and supporting technologies. However, it inherits "
            "broad and occasionally imprecise relationships from the graph. "
            "Its warnings about aspirational and scenario-based outcomes "
            "reduce the risk of treating them as achieved results."
        ),
    },

    12: {
        "source_grounding": "PARTIAL",
        "primary_classification": "CONTAINS_LLM_INTRODUCED_CLAIMS",
        "inherited_graph_error": "YES",
        "new_llm_claim": "YES",
        "numerical_claims_correct": "YES",
        "modality_preserved": "PARTIAL",
        "causal_language_problem": "NO",
        "unsupported_elaboration": "YES",
        "audit_notes": (
            "The principal EAG and PV targets are represented correctly, "
            "including 11 TWh by 2030 and the scenario-based 41 TWh potential "
            "by 2040. Some descriptions of financial deductions, EU building "
            "obligations and environmental legislation extend beyond what "
            "can safely be concluded from the cited graph relationships."
        ),
    },

    18: {
        "source_grounding": "PARTIAL",
        "primary_classification": "CONTAINS_LLM_INTRODUCED_CLAIMS",
        "inherited_graph_error": "YES",
        "new_llm_claim": "YES",
        "numerical_claims_correct": "NOT_APPLICABLE",
        "modality_preserved": "PARTIAL",
        "causal_language_problem": "YES",
        "unsupported_elaboration": "YES",
        "audit_notes": (
            "The report correctly identifies sector coupling, infrastructure, "
            "constraints and policy support as important themes. Statements "
            "such as PV having a massive influence on entire sectors and "
            "support measures enhancing deployment effectiveness are broader "
            "than the underlying evidence and sometimes imply causal effects."
        ),
    },

    19: {
        "source_grounding": "PARTIAL",
        "primary_classification": "CONTAINS_LLM_INTRODUCED_CLAIMS",
        "inherited_graph_error": "YES",
        "new_llm_claim": "YES",
        "numerical_claims_correct": "PARTIAL",
        "modality_preserved": "PARTIAL",
        "causal_language_problem": "NO",
        "unsupported_elaboration": "YES",
        "audit_notes": (
            "The report generally preserves planned and scenario-based status. "
            "However, it combines many separate strategy elements into one "
            "coherent implementation ecosystem and introduces interpretations "
            "about responsibilities, legal status and strategic levers that "
            "are not always directly supported."
        ),
    },

    9: {
        "source_grounding": "PARTIAL",
        "primary_classification": "UNSUPPORTED_ELABORATION",
        "inherited_graph_error": "YES",
        "new_llm_claim": "YES",
        "numerical_claims_correct": "YES",
        "modality_preserved": "PARTIAL",
        "causal_language_problem": "YES",
        "unsupported_elaboration": "YES",
        "audit_notes": (
            "The numerical values for 2022, 2023 and the 2040 potential are "
            "mostly correct. The statement that 2023 deployment matched or "
            "exceeded the planned long-term expansion pathway is an analytical "
            "comparison introduced by the report rather than an explicit "
            "source conclusion."
        ),
    },

    3: {
        "source_grounding": "PARTIAL",
        "primary_classification": "CONTAINS_LLM_INTRODUCED_CLAIMS",
        "inherited_graph_error": "YES",
        "new_llm_claim": "YES",
        "numerical_claims_correct": "YES",
        "modality_preserved": "PARTIAL",
        "causal_language_problem": "YES",
        "unsupported_elaboration": "YES",
        "audit_notes": (
            "The acceptance figures and the discussion of biodiversity and "
            "workforce initiatives are based on graph records. Nevertheless, "
            "the report combines heterogeneous subjects and occasionally turns "
            "qualitative associations into claims about demonstrated benefits, "
            "investment effects or increased adoption."
        ),
    },

    91: {
        "source_grounding": "HIGH",
        "primary_classification": "ACCURATE_SYNTHESIS",
        "inherited_graph_error": "NO",
        "new_llm_claim": "NO",
        "numerical_claims_correct": "NOT_APPLICABLE",
        "modality_preserved": "YES",
        "causal_language_problem": "NO",
        "unsupported_elaboration": "NO",
        "audit_notes": (
            "This is a narrow and cautious summary of the research, technology "
            "and innovation action field. It does not claim measured outcomes "
            "and explicitly acknowledges missing organizations, technologies "
            "and implementation evidence."
        ),
    },

    23: {
        "source_grounding": "PARTIAL",
        "primary_classification": "CONTAINS_LLM_INTRODUCED_CLAIMS",
        "inherited_graph_error": "YES",
        "new_llm_claim": "YES",
        "numerical_claims_correct": "YES",
        "modality_preserved": "PARTIAL",
        "causal_language_problem": "NO",
        "unsupported_elaboration": "YES",
        "audit_notes": (
            "The stakeholders and action fields are relevant and mostly "
            "source-supported. However, the report may attribute the 11 TWh "
            "target directly to the PV Strategy when its legal basis is the "
            "EAG, and it sometimes describes associated actors as necessary "
            "implementers without sufficiently direct evidence."
        ),
    },

    97: {
        "source_grounding": "PARTIAL",
        "primary_classification": "ACCURATE_SYNTHESIS",
        "inherited_graph_error": "YES",
        "new_llm_claim": "NO",
        "numerical_claims_correct": "NOT_APPLICABLE",
        "modality_preserved": "YES",
        "causal_language_problem": "NO",
        "unsupported_elaboration": "NO",
        "audit_notes": (
            "The report appropriately presents the ELWG measures as planned, "
            "proposed or under legislative review rather than implemented. "
            "Uncertainty remains because some underlying graph descriptions "
            "contain inconsistent legal-status information."
        ),
    },

    77: {
        "source_grounding": "PARTIAL",
        "primary_classification": "CAUSAL_GENERALIZATION",
        "inherited_graph_error": "YES",
        "new_llm_claim": "YES",
        "numerical_claims_correct": "NOT_APPLICABLE",
        "modality_preserved": "PARTIAL",
        "causal_language_problem": "YES",
        "unsupported_elaboration": "YES",
        "audit_notes": (
            "The source supports the existence of bureaucratic barriers and "
            "the intention to reduce them. The report goes further by stating "
            "that reduction efforts facilitate adoption and mitigate the "
            "constraint as an established causal effect, without outcome data."
        ),
    },

    81: {
        "source_grounding": "PARTIAL",
        "primary_classification": "CAUSAL_GENERALIZATION",
        "inherited_graph_error": "YES",
        "new_llm_claim": "YES",
        "numerical_claims_correct": "NOT_APPLICABLE",
        "modality_preserved": "PARTIAL",
        "causal_language_problem": "YES",
        "unsupported_elaboration": "YES",
        "audit_notes": (
            "Visibility and public acceptance are genuine source concepts. "
            "The report presents visibility as directly necessitating lasting "
            "acceptance, which is stronger causal language than the available "
            "qualitative and aspirational evidence supports."
        ),
    },

    43: {
        "source_grounding": "PARTIAL",
        "primary_classification": "CONTAINS_LLM_INTRODUCED_CLAIMS",
        "inherited_graph_error": "YES",
        "new_llm_claim": "YES",
        "numerical_claims_correct": "YES",
        "modality_preserved": "YES",
        "causal_language_problem": "NO",
        "unsupported_elaboration": "YES",
        "audit_notes": (
            "The principal 2022 and 2023 PV measurements and the scenario-based "
            "2040 values are represented correctly. The report appropriately "
            "distinguishes scenarios from achievements, but qualitative claims "
            "about predictable prices, energy sovereignty and supporting "
            "technologies are broader than the directly verified evidence."
        ),
    },
}

In [6]:
for community_id, values in community_audit.items():
    mask = community_report_review["community"] == community_id

    if mask.sum() != 1:
        print(
            f"Warning: community {community_id} matched "
            f"{mask.sum()} rows"
        )

    for column, value in values.items():
        community_report_review.loc[mask, column] = value

reviewed_reports = community_report_review[
    community_report_review["primary_classification"] != ""
]

print("Reviewed community reports:", len(reviewed_reports))

reviewed_reports[
    [
        "community",
        "title",
        "source_grounding",
        "primary_classification",
        "inherited_graph_error",
        "new_llm_claim",
        "numerical_claims_correct",
        "modality_preserved",
        "causal_language_problem",
        "unsupported_elaboration",
        "audit_notes",
    ]
]

Reviewed community reports: 13


,community,title,source_grounding,primary_classification,inherited_graph_error,new_llm_claim,numerical_claims_correct,modality_preserved,causal_language_problem,unsupported_elaboration,audit_notes
0,105,Austrian Renewable Expansion Act (EAG) – PV Ex...,HIGH,ACCURATE_SYNTHESIS,NO,NO,YES,YES,NO,NO,The EAG targets of 100% balance-sheet renewabl...
1,76,Photovoltaics and Systemic Expansion in Austri...,PARTIAL,ACCURATE_SYNTHESIS,YES,NO,NOT_APPLICABLE,YES,NO,NO,The report accurately synthesizes many PV poli...
2,12,"Support Schemes, Legal Targets, and Constraint...",PARTIAL,CONTAINS_LLM_INTRODUCED_CLAIMS,YES,YES,YES,PARTIAL,NO,YES,The principal EAG and PV targets are represent...
3,18,Photovoltaics and Sectoral Integration in the ...,PARTIAL,CONTAINS_LLM_INTRODUCED_CLAIMS,YES,YES,NOT_APPLICABLE,PARTIAL,YES,YES,The report correctly identifies sector couplin...
4,19,Austrian Photovoltaic Strategy and Its Ecosystem,PARTIAL,CONTAINS_LLM_INTRODUCED_CLAIMS,YES,YES,PARTIAL,PARTIAL,NO,YES,The report generally preserves planned and sce...
5,9,Austrian Photovoltaic Expansion Targets and Su...,PARTIAL,UNSUPPORTED_ELABORATION,YES,YES,YES,PARTIAL,YES,YES,"The numerical values for 2022, 2023 and the 20..."
6,3,"PV Acceptance, Biodiversity, and Workforce Dev...",PARTIAL,CONTAINS_LLM_INTRODUCED_CLAIMS,YES,YES,YES,PARTIAL,YES,YES,The acceptance figures and the discussion of b...
7,91,Austrian Photovoltaics Research and Innovation...,HIGH,ACCURATE_SYNTHESIS,NO,NO,NOT_APPLICABLE,YES,NO,NO,This is a narrow and cautious summary of the r...
8,23,"Stakeholders, Action Fields, and Strategic Aim...",PARTIAL,CONTAINS_LLM_INTRODUCED_CLAIMS,YES,YES,YES,PARTIAL,NO,YES,The stakeholders and action fields are relevan...
9,97,Regulatory Reform for PV Integration in Austri...,PARTIAL,ACCURATE_SYNTHESIS,YES,NO,NOT_APPLICABLE,YES,NO,NO,The report appropriately presents the ELWG mea...


In [7]:
print("Reviewed community reports:", len(reviewed_reports))

columns_to_summarize = [
    "source_grounding",
    "primary_classification",
    "inherited_graph_error",
    "new_llm_claim",
    "numerical_claims_correct",
    "modality_preserved",
    "causal_language_problem",
    "unsupported_elaboration",
]

for column in columns_to_summarize:
    print(f"\n{column}:")
    print(reviewed_reports[column].value_counts())

    print("\nPercentages:")
    print(
        reviewed_reports[column]
        .value_counts(normalize=True)
        .mul(100)
        .round(1)
    )

Reviewed community reports: 13

source_grounding:
source_grounding
PARTIAL    11
HIGH        2
Name: count, dtype: int64

Percentages:
source_grounding
PARTIAL    84.6
HIGH       15.4
Name: proportion, dtype: float64

primary_classification:
primary_classification
CONTAINS_LLM_INTRODUCED_CLAIMS    6
ACCURATE_SYNTHESIS                4
CAUSAL_GENERALIZATION             2
UNSUPPORTED_ELABORATION           1
Name: count, dtype: int64

Percentages:
primary_classification
CONTAINS_LLM_INTRODUCED_CLAIMS    46.2
ACCURATE_SYNTHESIS                30.8
CAUSAL_GENERALIZATION             15.4
UNSUPPORTED_ELABORATION            7.7
Name: proportion, dtype: float64

inherited_graph_error:
inherited_graph_error
YES    11
NO      2
Name: count, dtype: int64

Percentages:
inherited_graph_error
YES    84.6
NO     15.4
Name: proportion, dtype: float64

new_llm_claim:
new_llm_claim
YES    9
NO     4
Name: count, dtype: int64

Percentages:
new_llm_claim
YES    69.2
NO     30.8
Name: proportion, dtype: flo

# Pilot 2 Community-Report Audit

## Objective

This inspection evaluated whether the community reports generated by
GraphRAG faithfully summarized the underlying Austrian Photovoltaic
Strategy and its extracted knowledge graph.

Community reports must not be treated as source evidence. They are
LLM-generated interpretations constructed from extracted entities,
relationships, and community structures.

## Audit design

Pilot 2 produced 110 community reports, compared with 62 reports in
Pilot 1.

This represents:

- an absolute increase of 48 reports;
- a relative increase of 77.4%.

A systematic sample of 13 Pilot 2 community reports was inspected. The
sample combined:

- the five highest-ranked reports;
- the five reports representing the largest communities;
- seven reproducibly selected random reports;
- removal of duplicate selections.

Each sampled report was assessed for:

- source grounding;
- accuracy of synthesis;
- inherited graph errors;
- claims newly introduced by the report-generating LLM;
- correctness of numerical claims;
- preservation of modality and implementation status;
- unsupported causal language;
- unsupported elaboration.

## Results

### Source grounding

| Grounding classification | Reports | Percentage |
|---|---:|---:|
| HIGH | 2 | 15.4% |
| PARTIAL | 11 | 84.6% |
| LOW | 0 | 0.0% |

All sampled reports contained information connected to the source or
the extracted graph. However, only 15.4% were assessed as highly
grounded throughout. Most reports combined valid source information
with broader interpretations derived from the graph.

### Primary reliability classification

| Classification | Reports | Percentage |
|---|---:|---:|
| ACCURATE_SYNTHESIS | 4 | 30.8% |
| CONTAINS_LLM_INTRODUCED_CLAIMS | 6 | 46.2% |
| CAUSAL_GENERALIZATION | 2 | 15.4% |
| UNSUPPORTED_ELABORATION | 1 | 7.7% |

Only 4 of the 13 reports were primarily classified as accurate
syntheses.

The most common problem was the introduction of claims or
interpretations that were not explicitly present in the source or
could not be safely derived from the graph.

### Inherited graph errors

| Inherited graph error | Reports | Percentage |
|---|---:|---:|
| YES | 11 | 84.6% |
| NO | 2 | 15.4% |

Most community reports inherited at least one problem from the entity
or relationship graph.

These problems included:

- duplicate or unresolved entities;
- broad or imprecise relationship types;
- incorrect or ambiguous relationship endpoints;
- associations presented with greater specificity than the graph
  evidence allowed;
- inconsistent implementation-status information.

This shows that community-report quality is strongly dependent on the
quality of the preceding graph-extraction and entity-resolution
stages.

### New LLM-introduced claims

| New LLM claim | Reports | Percentage |
|---|---:|---:|
| YES | 9 | 69.2% |
| NO | 4 | 30.8% |

Nine reports contained at least one interpretive claim introduced
during report generation.

Therefore, community-report errors cannot be attributed solely to the
underlying graph. The report-generation LLM represents an additional
interpretation layer that may introduce new generalizations.

### Numerical claims

| Numerical assessment | Reports | Percentage |
|---|---:|---:|
| YES | 6 | 46.2% |
| PARTIAL | 1 | 7.7% |
| NOT_APPLICABLE | 6 | 46.2% |

Among the seven reports containing relevant numerical claims, six
presented the principal numerical values correctly and one was only
partially correct.

This indicates that Pilot 2 substantially improved the representation
of targets and market metrics. Important values such as the following
were generally retained:

- at least 11 TWh of additional PV generation by 2030;
- approximately 41 TWh of PV potential by 2040;
- approximately 2.5 GW of additional PV capacity in 2023;
- approximately 6.3 TWh of PV generation in 2023;
- approximately 10% of electricity consumption supplied by PV in
  2023.

Nevertheless, a correct number does not guarantee a correct claim. The
status, responsible policy, comparison baseline, deadline, and
relationship connecting the value to other entities must also be
verified.

### Preservation of modality

| Modality assessment | Reports | Percentage |
|---|---:|---:|
| YES | 5 | 38.5% |
| PARTIAL | 8 | 61.5% |

Only five reports consistently preserved distinctions such as:

- measured outcome;
- explicit fact;
- target;
- scenario;
- recommendation;
- planned action;
- proposal;
- legal requirement.

Eight reports preserved these distinctions only partially. In these
cases, the reports sometimes mixed established measures, proposals,
recommendations, scenarios, and aspirations within the same narrative.

### Causal-language problems

| Causal-language problem | Reports | Percentage |
|---|---:|---:|
| YES | 5 | 38.5% |
| NO | 8 | 61.5% |

Five reports used language suggesting that one factor caused,
facilitated, increased, mitigated, or enabled another outcome without
sufficient outcome evidence.

For example, an intended reduction of bureaucratic barriers may be
described in the source as a proposed measure. A community report
should not automatically state that the measure has already increased
PV deployment.

### Unsupported elaboration

| Unsupported elaboration | Reports | Percentage |
|---|---:|---:|
| YES | 9 | 69.2% |
| NO | 4 | 30.8% |

Unsupported elaboration appeared in nine reports. Typical examples
included:

- combining separate statements into an apparently established causal
  narrative;
- assigning responsibilities to actors based only on association;
- interpreting policy intentions as implementation effects;
- treating a broad relationship as a specific operational mechanism;
- expanding limited qualitative evidence into general market
  conclusions.

## Interpretation

Pilot 2 community reports are more domain-specific and more useful
than the Pilot 1 reports. The controlled schema allows reports to
distinguish policies, technologies, targets, market metrics,
infrastructure, support schemes, stakeholders, and constraints.

The reports also show improved caution. Many explicitly acknowledge
that targets are scenario-based, proposed, planned, or unverified.

However, community reports remain the least authoritative stage of
the pipeline because they are produced after several transformations:

1. PDF conversion;
2. chunking;
3. entity extraction;
4. entity typing;
5. relationship extraction;
6. community detection;
7. LLM report generation.

An error or ambiguity introduced at an earlier stage can therefore
propagate into the community report. The report-generating LLM can
then introduce an additional layer of interpretation.

## Main conclusion

Pilot 2 community reports are suitable for:

- exploratory analysis;
- identifying major themes;
- providing an overview of graph communities;
- supporting question answering;
- locating potentially relevant entities and relationships.

They are not independently suitable for:

- factual verification;
- establishing causality;
- confirming policy implementation;
- proving measured market effects;
- serving as authoritative evidence without source tracing.

The original PDF and its text units must remain the authoritative
evidence. Every important generated claim should be traceable through:

`Community report → Relationship/entity → Text unit → Source document`

## Recommended treatment in the controlled KG

Community reports should not be imported as verified facts into the
controlled knowledge graph.

They should instead be stored as generated analytical artifacts with
metadata such as:

- source community identifier;
- generation model;
- generation date;
- grounding status;
- verification status;
- inherited-error indicator;
- modality-preservation assessment;
- source entity and relationship references.

Verified entities, relationships, targets, metrics, and policy claims
should remain separate from LLM-generated narrative summaries.

## Recommended improvements

1. Perform entity resolution before community detection and report
   generation.

2. Remove or correct unsupported and incorrectly directed
   relationships before generating reports.

3. Require report prompts to preserve explicit modality labels.

4. Prohibit causal language unless the supporting relationship is
   explicitly causal and source-supported.

5. Require every important claim to cite graph records and their
   supporting text units.

6. Distinguish measured outcomes from targets, scenarios,
   recommendations, proposals, and legal requirements.

7. Treat community-report rank as GraphRAG importance metadata, not as
   factual confidence.

8. Use community reports for navigation and synthesis, while using
   source-backed graph records for formal analysis and explanation.

## Overall assessment

Pilot 2 substantially improves the thematic usefulness of GraphRAG's
community reports, especially for policy, technical, and market
analysis. Nevertheless, the audit shows that only 30.8% of the sampled
reports were primarily accurate syntheses, while 69.2% contained new
LLM claims and 69.2% contained unsupported elaboration.

The main bottleneck is therefore no longer simply generic entity
typing. It is now the propagation of graph imperfections into
community reports, combined with additional interpretation during
LLM-based report generation.